test/scenarios.py

```diff
@@ -5149,11 +5149,11 @@
                         )
                     ),
                     action.Wait(event.Start(0, FILES["testfile-big"].id)),
-                    action.Wait(
-                        event.FinishFileCanceled(0, FILES["testfile-big"].id, True)
-                    ),
-                    action.Wait(
-                        event.FinishFileRejected(0, FILES["testfile-big"].id, True)
+                    action.WaitRacy(
+                        [
+                            event.FinishFileRejected(0, FILES["testfile-big"].id, True),
```

Comment:
The thing is that they probably should be emitted in order. As I think of it more, we might skip the `Canceled` event here like not it tests but in code. It's completely redundant for the API user.
Let's merge this one and think of it later, possibly creating a ticket